opentelemetry-stdout/src/logs/exporter.rs

```diff
@@ -1,38 +1,24 @@
 use async_trait::async_trait;
+use chrono::{DateTime, Utc};
 use core::fmt;
-use opentelemetry::{
-    logs::{LogError, LogResult},
-    ExportError,
-};
 use opentelemetry_sdk::export::logs::{ExportResult, LogData};
 use opentelemetry_sdk::Resource;
-use std::borrow::Cow;
-use std::io::{stdout, Write};
+use std::{borrow::Cow, sync::atomic};
 
-type Encoder =
-    Box<dyn Fn(&mut dyn Write, crate::logs::transform::LogData) -> LogResult<()> + Send + Sync>;
-
-/// A [`LogExporter`] that writes to [`Stdout`] or other configured [`Write`].
-///
-/// [`LogExporter`]: opentelemetry_sdk::export::logs::LogExporter
-/// [`Write`]: std::io::Write
-/// [`Stdout`]: std::io::Stdout
+/// An OpenTelemetry exporter that writes Logs to stdout on export.
 pub struct LogExporter {
-    writer: Option<Box<dyn Write + Send + Sync>>,
-    encoder: Encoder,
     resource: Resource,
-}
-
-impl LogExporter {
-    /// Create a builder to configure this exporter.
-    pub fn builder() -> LogExporterBuilder {
-        Default::default()
-    }
+    is_shutdown: atomic::AtomicBool,
+    resource_emitted: bool,
```

Comment:
Also, what's the reason for printing resource information only once? I think it should be okay to keep printing the resource with every export for easy reference. Unlike the collector, `stdout` of the user application could be used for printing a lot of other things, right?

Reply:
Resource is fixed and won't change.. So I wasn't sure if its worth showing the same info repeatedly! (It kinda pollutes the output, so maybe we can make an opt-in flag to keep printing Resource)

Reply:
Also - for OTel collector, resource can be anything as there could be  any number of apps sending to Collector. (It still tried to print Resource once per batch)

Reply:
Once per batch might offer a better user experience than ONLY once throughout the application lifetime. It all depends on the usage pattern. If a user is emitting a lot of other stuff to stdout and also testing multiple batches of export, then they have to always scroll up on the console to where the resource info was first printed.

Reply:
Issue with **Once per batch** is that, the default pairing of stdout exporter is simple processor, so the batch has only one item, making it effectively emit Resource with each item!